### PR TITLE
FXIOS-999 ⁃ Simpler Photo handling

### PR DESCRIPTION
**Pull request by chrmod**
Based on https://github.com/mozilla-mobile/firefox-ios/pull/7424

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-999)

### CODE REVIEW

**Comment by chrmod** [comment]
Out of curiosity. How it is possible that firefox-ios is able be a default browser if:
* `NSPhotoLibraryUsageDescription` was removed from Info.plist
* `PHPhotoLibrary.requestAuthorization` is still present in the codebase

Builds from the forks I maintain are automatically rejected due to `ITMS-90683: Missing Purpose String in Info.plist`

**Comment by chrmod** [comment]
Please update commit massage on merge. Wrong PR was mentioned. It should be https://github.com/mozilla-mobile/firefox-ios/pull/7424/files

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -2227,44 +2227,10 @@ extension BrowserViewController: ContextMenuHelperDelegate {
...
-                            return
-                        }
-
-                        handlePhotoLibraryAuthorized()
-                    })
-                } else if photoAuthorizeStatus == .authorized {
-                    handlePhotoLibraryAuthorized()
-                } else {
-                    handlePhotoLibraryDenied()
+                self.getImageData(url) { data in
```
Comment (nbhasin2) [inline Client/Frontend/Browser/BrowserViewController.swift:2231]: Thanks, @chrmod

This looks good although I'd request some changes in regards to how we handle when user has not granted permission. 

Below is some work I did in a similar way but that handles when user is unable to save pic. Could I request modification to this PR in same as -> https://github.com/mozilla-mobile/firefox-ios/pull/7424/files

**Review by nbhasin2 — CHANGES_REQUESTED** [review]

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 1 above)
Comment (chrmod) [reply Client/Frontend/Browser/BrowserViewController.swift:2231]: Hi @nbhasin2 

This is already using methods defined in `BrowserViewController+WebViewDelegates.swift`.

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**